core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -531,4 +538,23 @@ protected static Fuzziness randomFuzziness(String fieldName) {
     protected static boolean isNumericFieldName(String fieldName) {
         return INT_FIELD_NAME.equals(fieldName) || DOUBLE_FIELD_NAME.equals(fieldName);
     }
+
+    protected static class MockTermsLookupFetchService extends TermsLookupFetchService {
+
+        private static List<Object> randomTerms = new ArrayList<>();
+
+        public MockTermsLookupFetchService() {
+            super(null, Settings.Builder.EMPTY_SETTINGS);
+            randomTerms.addAll(Lists.newArrayList(generateRandomStringArray(10, 10, false, randomBoolean())));
```

Comment:
out of curiosity, what happens if we allow null here? it can happen that a document has null values for a field no?